numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -656,6 +656,26 @@ npy_updateifcopy_deprecation(PyObject* NPY_UNUSED(self), PyObject* args)
     Py_RETURN_NONE;
 }
 
+/* used to test PyArray_As1D usage emits not implemented error */
+static PyObject*
+npy_pyarrayas1d_deprecation(PyObject* NPY_UNUSED(self), PyObject* NPY_UNUSED(args))
+{
+    PyObject *arg1;
+    double arg2[2] = {1, 2};
+    PyObject* temp = PyArray_As1D((PyObject **)&arg1, (char **)&arg2, (int *)4, NPY_DOUBLE);
+    return NULL;
+}
+
+/* used to test PyArray_As2D usage emits not implemented error */
+static PyObject*
+npy_pyarrayas2d_deprecation(PyObject* NPY_UNUSED(self), PyObject* NPY_UNUSED(args))
+{
+    PyObject *arg1;
+    double arg2[2][2] = {{1, 2}, {3, 4}};
+    PyObject* temp = PyArray_As2D((PyObject **)&arg1, (char ***)&arg2, (int *)4, (int *)6, NPY_DOUBLE);
+    return NULL;
+}
+
```

Comment:
the compiler is complaining that you never use `temp`. Either return it or call `Py_DECREF(temp)`

Reply:
The interface to `PyArray_AsCArray` is very strange - it modifies the first argument, replacing it with the result of `PyArray_FromAny` but only if it returns success. In the tests this doesn't matter, since the first argument is a valid ndarray so the value returned from `PyArray_FromAny` is the same as the input, but in these tests (if successful) it would have changed `result`.

All that is irrelevant since the deprecated functions fail, but I tried to use a correct approach for the unused branch anyway.